pkg/policy/api/l4.go

```diff
@@ -55,6 +55,34 @@ func (p PortProtocol) Covers(other PortProtocol) bool {
 	return true
 }
 
+// K8sSecret is a reference to a k8s secret.
+type K8sSecret struct {
+	// Namespace is the k8s namespace in which the secret exists. If
+	// namespace is omitted, the namespace of the enclosing rule is assumed.
+	Namespace string `json:"namespace,omitempty"`
+
+	// Name is the name of the k8s secret
+	Name string `json:"name,omitempty"`
+}
+
+// TLSContext provides TLS configuration via reference to k8s secrets.
+type TLSContext struct {
+	// Certificate is a chain of certificates presented to the remote party.
+	// If specified for an originating TLS context, then this is used as a
+	// client certificate
+	Certificate K8sSecret `json:"certificate,omitempty"`
+
+	// PrivateKey is the private key used to encrypt the TLS messages. This
+	// is the private part of the public/private key pair, corresponding to
+	// the public key in the certificate.
+	PrivateKey K8sSecret `json:"privatekey,omitempty"`
```

Comment:
`private-key`

Reply:
This (and others) are for the json annotation, right?

Reply:
yes :-)